### ISSUE TITLE
[10.10.2] PacketTooBigException

### ISSUE THREAD

**Issue by Teslicek** [kangarko/ChatControl#1781]
After updating to 10.10.2 and using MySQL database, I got an insane spam in my console after a restart (because lots of players joined at the same time and database was not fast enough or something and I also had a big data.db file before). And it made ChatControl go crazy and throw so many errors, which resulted in a 600MB+ errors.txt file, which I had to remove and then it was making another one (around 60MB), until I killed the server process. and it looked like this:
```
------------------------------------[ 12.03.2022 14:52:32 ]-----------------------------------
ChatControlRed 10.10.2 encountered a PacketTooBigException
Running Pufferfish 1.18.2-R0.1-SNAPSHOT and Java 17.0.2
Plugins: (hidden), ChatControlRed v10.10.2
----------------------------------------------------------------------------------------------

More Information: 
(this line is so long its actually about 60MB, but I took just some of it)
Error on updating MySQL with: INSERT INTO ChatControl (UUID, Name, Nick, Data, LastModified) VALUES ('4968e689-84b4-435a-aaad-e65adb96ad78', 'Doctor_DHAN', NULL, '{"Players":{"9a76dd66-a830-4d4a-9f72-80b0789cc3e8":{"Name":"Txslx","UUID":"9a76dd66-a830-4d4a-9f72-80b0789cc3e8","Channels":{"standard":"write"},"Channels.standard":"write","Last_Active":"1647090977930","Spying_Channels":["standard"],"Tags":{"nick":"§bTxslx"},"Tags.nick":"§bTxslx"},"9a76dd66-a830-4d4a-9f72-80b0789cc3e8.Name":"Txslx","9a76dd66-a830-4d4a-9f72-80b0789cc3e8.UUID":"9a76dd66-a830-4d4a-9f72-80b0789cc3e8","9a76dd66-a830-4d4a-9f72-80b0789cc3e8.Channels":{"standard":"write"},"9a76dd66-a830-4d4a-9f72-80b0789cc3e8.Channels.standard":"write","9a76dd66-a830-4d4a-9f72-80b0789cc3e8.Last_Active":"1647090977930","9a76dd66-a830-4d4a-9f72-80b0789cc3e8.Spying_Channels":["standard"],"9a76dd66-a830-4d4a-9f72-80b0789cc3e8.Tags":{"nick":"§bTxslx"},"9a76dd66-a830-4d4a-9f72-80b0789cc3e8.Tags.nick":"§bTxslx","bcd6552b-4efb-49cd-b8ef-4776700792fd":{"Name":"Nextinka","UUID":"bcd6552b-4efb-49cd-b8ef-4776700792fd","Last_Active":"1647092014601","Spying_Channels":["standard"],"Tags":{"nick":"§x§7§c§f§b§f§bN§x§7§a§f§b§f§2e§x§7§8§f§b§e§ax§x§7§6§f§c§e§1t§x§7§4§f§c§d§8i§x§7§1§f§c§d§0n§x§6§f§f§c§c§7k§x§6§d§f§d§b§ea"},"Tags.nick":"§x§7§c§f§b§f§bN§x§7§a§f§b§f§2e§x§7§8§f§b§e§ax§x§7§6§f§c§e§1t§x§7§4§f§c§d§8i§x§7§1§f§c§d§0n§x§6§f§f§c§c§7k§x§6§d§f§d§b§ea","Channels":{"standard":"write"},"Channels.standard":"write"},"bcd6552b-4efb-49cd-b8ef-4776700792fd.Name":"Nextinka","bcd6552b-4efb-49cd-b8ef-4776700792fd.UUID":"bcd6552b-4efb-49cd-b8ef-4776700792fd","bcd6552b-4efb-49cd-b8ef-4776700792fd.Last_Active":"1647092014601","bcd6552b-4efb-49cd-b8ef-4776700792fd.Spying_Channels":["standard"],"bcd6552b-4efb-49cd-b8ef-4776700792fd.Tags":{"nick":"§x§7§c§f§b§f§bN§x§7§a§f§b§f§2e§x§7§8§f§b§e§ax§x§7§6§f§c§e§1t§x§7§4§f§c§d§8i§x§7§1§f§c§d§0n§x§6§f§f§c§c§7k§x§6§d§f§d§b§ea"},"bcd6552b-4efb-49cd-b8ef-4776700792fd.Tags.nick":"§x§7§c§f§b§f§bN§x§7§a§f§b§f§2e§x§7§8§f§b§e§ax§x§7§6§f§c§e§1t§x§7§4§f§c§d§8i§x§7§1§f§c§d§0n§x§6§f§f§c§c§7k§x§6§d§f§d§b§ea","bcd6552b-4efb-49cd-b8ef-4776700792fd.Channels":{"standard":"write"},"bcd6552b-4efb-49cd-b8ef-4776700792fd.Channels.standard":"write","c76c9539-193d-489b-b60d-2910ab4c7777":{"Name":"_MrWhite_","UUID":"c76c9539-193d-489b-b60d-2910ab4c7777","Channels":{"standard":"write"},"Channels.standard":"write","Last_Active":"1647092006261","Spying_Channels":["standard"],"Tags":{"nick":"§c_MrWhite_"},"Tags.nick":"§c_MrWhite_"},"c76c9539-193d-489b-b60d-2910ab4c7777.Name":"_MrWhite_","c76c9539-193d-489b-b60d-2910ab4c7777.UUID":"c76c9539-193d-489b-b60d-2910ab4c7777","c76c9539-193d-489b-b60d-2910ab4c7777.Channels":{"standard":"write"},"c76c9539-193d-489b-b60d-2910ab4c7777.Channels.standard":"write","c76c9539-193d-489b-b60d-2910ab4c7777.Last_Active":"1647092006261","c76c9539-193d-489b-b60d-2910ab4c7777.Spying_Channels":["standard"],"c76c9539-193d-489b-b60d-2910ab4c7777.Tags":{"nick":"§c_MrWhite_"},"c76c9539-193d-489b-b60d-2910ab4c7777.Tags.nick":"§c_MrWhite_","82381c8b-2c3f-47ea-ab5c-5d16c3f9c8d6":{"Name":"MeowGamer3308","UUID":"82381c8b-2c3f-47ea-ab5c-5d16c3f9c8d6","Channels":{"standard":"write"},"Channels.standard":"write","Last_Active":"1647092798520"},"82381c8b-2c3f-47ea-ab5c-5d16c3f9c8d6.Name":"MeowGamer3308","82381c8b-2c3f-47ea-ab5c-5d16c3f9c8d6.UUID":"82381c8b-2c3f-47ea-ab5c-5d16c3f9c8d6","82381c8b-2c3f-47ea-ab5c-5d16c3f9c8d6.Channels":{"standard":"write"},"82381c8b-2c3f-47ea-ab5c-5d16c3f9c8d6.Channels.standard":"write","82381c8b-2c3f-47ea-ab5c-5d16c3f9c8d6.Last_Active":"1647092798520","7e17fa6a-214d-4580-acb6-6c0a60bad959":{"Name":"sxvitarr","UUID":"7e17fa6a-214d-4580-acb6-6c0a60bad959","Channels":{"standard":"write"},"Channels.standard":"write","Last_Active":"1647073334033"},"7e17fa6a-214d-4580-acb6-6c0a60bad959.Name":"sxvitarr","7e17fa6a-214d-4580-acb6-6c0a60bad959.UUID":"7e17fa6a-214d-4580-acb6-6c0a60bad959","7e17fa6a-214d-4580-acb6-6c0a60bad959.Channels":{"standard":"write"},"7e17fa6a-214d-4580-acb6-6c0a60bad959.Channels.standard":"write","7e17fa6a-214d-4580-acb6-6c0a60bad959
(and then it continues like this)
PacketTooBigException Packet for query is too large (20,894,102 > 16,777,216). You can change this value on the server by setting the 'max_allowed_packet' variable.
	 at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:107)
	 at com.mysql.cj.jdbc.StatementImpl.executeUpdateInternal(StatementImpl.java:1340)
	 at com.mysql.cj.jdbc.StatementImpl.executeLargeUpdate(StatementImpl.java:2089)
	 at com.mysql.cj.jdbc.StatementImpl.executeUpdate(StatementImpl.java:1251)
	 at com.zaxxer.hikari.pool.ProxyStatement.executeUpdate(ProxyStatement.java:119)
	 at com.zaxxer.hikari.pool.HikariProxyStatement.executeUpdate(HikariProxyStatement.java)
	 at ChatControl-Red-10.10.2.jar//org.mineacademy.chatcontrol.lib.database.SimpleDatabase.update(SimpleDatabase.java:337)
	 at ChatControl-Red-10.10.2.jar//org.mineacademy.chatcontrol.lib.database.SimpleDatabase.insert(SimpleDatabase.java:277)
	 at ChatControl-Red-10.10.2.jar//org.mineacademy.chatcontrol.lib.database.SimpleDatabase.insert(SimpleDatabase.java:263)
	 at ChatControl-Red-10.10.2.jar//org.mineacademy.chatcontrol.model.Database.saveCache(Database.java:393)
	 at ChatControl-Red-10.10.2.jar//org.mineacademy.chatcontrol.PlayerCache.lambda$save$1(PlayerCache.java:333)
	 at org.bukkit.craftbukkit.v1_18_R2.scheduler.CraftTask.run(CraftTask.java:101)
	 at org.bukkit.craftbukkit.v1_18_R2.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57)
	 at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22)
	 at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	 at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	 at java.base/java.lang.Thread.run(Thread.java:833)
----------------------------------------------------------------------------------------------
```
I have no idea how to fix it, but I got it only to work, because I switched back to using data.db file instead of a mysql database.

**Comment by ElBananaa** [comment]
Hey,
@kangarko it seems like the changes you made in the previous version may be causing this issue.
With the new implementation of HikariCP spliting heavy db requests would be nice as it could avoid such issues without causing any performance issues ^^

@Teslicek Please make a backup of your database and then try to completely reset it to see if it helps.
EDIT: It appears what I suggested right above, didn't help, so forget about it.

**Comment by ElBananaa** [comment]
We managed to "fix" the user's issue by increasing the `max_allowed_packet` config option in `/etc/mysql/mariadb.conf.d/50-server.cnf` to `512M`. However, users shouldn't have to do this in order to fix this issue. Some may actually not even be able to do this (Only users with a dedicated server or a VPS will be able to do so).

**Comment by itsBeaco** [comment]
Experiencing the same issue

**Comment by ElBananaa** [comment]
Related to #1784

**Comment by kangarko** [comment]
Thanks, this will be fixed as a part of #1784 hotfix, apology for the delay. I recommend to remove the ChatControl table completely to start fresh.
